### ISSUE TITLE
Move `dim` implementation from schemes to algebras

### ISSUE THREAD

**Issue by lgoettgens** [oscar-system/Oscar.jl#2771]
> One comment beforehand: Most of this should not happen on the geometric side, but on the algebra side. We need all these fixes on the algebra side as well and should only have them in one spot. The geometric side should directly refer `dim(X)` to `dim(OO(X))` in the local and affine cases.

_Originally posted by @afkafkafk13 in https://github.com/oscar-system/Oscar.jl/pull/2766#discussion_r1315146328_

**Comment by afkafkafk13** [comment]
Just to avoid forgetting it: 
#2860  had to work around this and the workaround needs to be removed, once this issue has been is solved.

**Comment by simonbrandhorst** [comment]
@afkafkafk13  we have dim for rings . Can you proceed?

**Comment by afkafkafk13** [comment]
Yes, I can take it from here and do the clean-ups this afternoon.
